DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,211 @@
+# Development Guide
+
+The following document will be a reference guide to coding standards of cp-ansible. It is written to be idempotent, meaning whatever you set in your inventory file, the playbook and roles should provision on your hosts, and can be run multiple times without causing unnecessary changes to those hosts.
+
+#### Table Of Contents
+
+[Roles](#roles)
+
+[Variables](#variables)
+
+[Component Properties](#component-properties)
+
+[Custom Filters](#custom-filters)
+
+[Testing](#testing)
+
+[Branching Model](#branding-model)
+
+## Roles
+Each Confluent component has its own role, with the name `confluent.<component_name>`. Within that role, `main.yml` is the entrypoint of all tasks run when the role is invoked. Here are a few tasks and coding standards associated with them:
+
+```
+- name: Write Service Overrides	(1)
+  template:	(2)
+    src: override.conf.j2
+    dest: "{{ kafka_broker.systemd_override }}"	(3)
+    mode: 0640	(4)
+    owner: "{{kafka_broker_user}}"	(5)
+    group: "{{kafka_broker_group}}"
+  notify: restart kafka	(6)
+```
+
+1. A name clearly defining what the task accomplishes, using capital letters
+2. Uses an idempotent ansible module whenever possible
+3. Make use of variables instead of hard coding paths
+4. For file creation use 0640 permission, for directory creation use 0750 permission. There are some exceptions, but be sure to secure files.
+5. Proper ownership set
+6. Trigger component restart handler when necessary
+
+```
+- name: Register Cluster
+  include_tasks: register_cluster.yml	(1)
+  when: kafka_connect_cluster_name|length > 0	(2)
+```
+
+1. Put sets of related tasks into their own task file and include them (not import)
+2. Use conditionals to skip entire sets of tasks, or singular tasks
+
+```
+- name: Import the CA cert into the Keystore
+  shell: |
```

Comment:
Unfortunately that standard wont be seen if you look thru the code, so we can update all the pipes and then call it out as a standard. I personally like the pipes because it feels more like a bash script. All in all i don't think this bit is false because it only documents the no_log standard

Reply:
That's fine! I just think it is less error-prone to avoid the pesky `\` and end-of-line. But personal preference is not something to discuss. 😄